graphql_compiler/compiler/emit_sql.py

```diff
@@ -103,6 +108,53 @@ def _find_folded_outputs(ir):
     return folded_outputs
 
 
+class XMLPathBinaryExpression(BinaryExpression):
+    pass
+
+
+@compiles(_CompileLabel, 'default')
+def compile_xmlpath(element, compiler, **kw):
+    """Suppress labeling when compiling XML PATH subqueries, otherwise compile as usual."""
+    if type(element.element) == XMLPathBinaryExpression:
+        # this indicates that no label should be inserted for this element
+        kw.update(
+            within_columns_clause=False
+        )
+    return compiler.visit_label(element, **kw)
+
+
+def agg_table(output_column, where):
+    """Produce an MSSQL 2014-style XML PATH-based aggregation subquery.
+
+    Args:
+        output_column: SQLAlchemy Column, value being aggregated with XML PATH
+        where: SQLAlchemy BinaryExpression, predicate used to filter
+    Returns:
+        An SQLAlchemy Selectable corresponding to an XML PATH subquery
+        which aggregates the values of the output_column from the output vertex.
+        Joining from the preceding vertex to the correct output vertex
+        is performed with the WHERE clause.
+    """
+    xml_column = (u'~|*' + func.REPLACE(output_column, u'|', u'||'))
+
+    # as a custom type, you can't directly construct an XMLPathBinaryExpression from plain text
+    xml_column = XMLPathBinaryExpression(xml_column.left, xml_column.right, xml_column.operator)
+
+    # REPLACEs to undo autoconversion of HTML special characters to entity reference shorthand
```

Comment:
I forgot to carry over this comment from #555 (in part because I forgot the details - this goes to show you how scary it really is!). I'm not sure if we ever debated the costs/benefits of this design trade-off.

```
I have edited the plan to defer all replacement of auto-escaped chars to their original form until python post processing. The reason why is that in addition to convering <,>,& to &lt;, etc. there is also a whole host of special characters are converted to entity numbers, e.g. &#0xHEX; and it would be best to capture those with one single regex - easier done in python than in mssql. A bit more info is here https://www.w3schools.com/html/html_entities.asp

An important design alternative here is that it is possible to avoid the auto-escaping if we use XML PATH(''), TYPE).value('.', 'VARCHAR(max)') - the only downside is that this form isn't robust to inputs containing certain unicode characters and would need to be both preprocessed to pass through the TYPE function and postprocessed to undo that preprocessing. b/c it is robust to all character inputs (to my knowledge), the method using just XML PATH('') only needs postprocessing.
```

Reply:
If we defer all of the substitution of the hex characters to post processing, it might be smart to do so for the other characters as well. Then I can just make a list on this method of all the reverse mappings that need to be done and what order to do them in.

Reply:
Btw I think https://docs.python.org/3/library/html.html#html.unescape should do the right conversions from XML entity encoding into proper text.

Reply:
I'll put this link in a comment on how post-processing should work.